scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+import functools
+
+import numpy as np
+from scipy._lib._array_api import array_namespace, is_cupy, is_torch, is_numpy
+from . import _ufuncs
+# These don't really need to be imported, but otherwise IDEs might not realize
+# that these are defined in this file / report an error in __init__.py
+from ._ufuncs import (
+    log_ndtr, ndtr, ndtri, erf, erfc, i0, i0e, i1, i1e,  # noqa
+    gammaln, gammainc, gammaincc, logit, expit)  # noqa
+
+_SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", False)
+array_api_compat_prefix = "scipy._lib.array_api_compat.array_api_compat"
+
+
+def get_array_special_func(f_name, xp, n_array_args):
+    if is_numpy(xp):
+        f = getattr(_ufuncs, f_name, None)
+    elif is_torch(xp):
+        f = getattr(xp.special, f_name, None)
+    elif is_cupy(xp):
+        import cupyx  # type: ignore[import]
+        f = getattr(cupyx.scipy.special, f_name, None)
+    elif xp.__name__ == f"{array_api_compat_prefix}.jax":
+        f = getattr(xp.scipy.special, f_name, None)
+    else:
+        f_scipy = getattr(_ufuncs, f_name, None)
+        def f(*args, **kwargs):
+            array_args = args[:n_array_args]
+            other_args = args[n_array_args:]
+            array_args = [np.asarray(arg) for arg in array_args]
+            out = f_scipy(*array_args, *other_args, **kwargs)
+            return xp.asarray(out)
+
+    return f
+
+# functools.wraps doesn't work because:
+# 'numpy.ufunc' object has no attribute '__module__'
+def support_cupy_torch_jax(f_name, n_array_args):
```

Comment:
I'm not sure about the name `support_cupy_torch_jax`. At the moment, jax isn't even actually supported and presumably in the future there will be other backends to support. Maybe something more generic like `support_alternative_backends`?

Reply:
I was following Ralf's suggestion here: https://github.com/scipy/scipy/pull/19023#discussion_r1286908503.

Reply:
I see. I like `support_alternative_backends` better, and it looks like there's support from @lucascolley too. Let's see what @rgommers has to say I guess.

Reply:
I'll just change it. My impression was that the original `dispatch` was not descriptive enough, and `support_alternative_backends` is also more descriptive than that.